pyglet/window/__init__.py

```diff
@@ -1154,7 +1154,7 @@ def set_vsync(self, vsync):
                 If True, vsync is enabled, otherwise it is disabled.
 
         """
-        raise NotImplementedError('abstract')
+        self._vsync = vsync
```

Comment:
I could include:
```python
if self._context:
    self._context.set_vsync(vsync)
```
but there are a few questions:
1) Is checking for existing of self._context necessary. Only CocoaWindow performs this check
2) Is it a good idea to include the extra logic here

Reply:
There seem to be a lot of similar checks in the CocoaWindow, but I'm not sure if they're really necessary. The context is set during `__init__`, so this shouldn't be any different from other Windows. I think we should try to confirm on a Mac.

Reply:
I'll test it on my mac in a bit

Reply:
Context seems to be there from my testing on macOS. Can't see anything in the code that would suggest it wouldn't be there, so I'm going to remove the check for it